promtimer/annotations.py

```diff
@@ -0,0 +1,166 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from os import path
+import time
+import json
+from dateutil import parser as dateparser
+import urllib.request
+
+ANNOTS_URL = 'http://localhost:13300/api/annotations'
+ANNOTS_HEADERS = {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+}
+ANNOTS_EVENTS_START = {
+    'rebalance_start': 'rebalance',
+    'failover_start': 'failover',
+}
+ANNOTS_EVENTS_END = {
+    'rebalance_finish': 'rebalance',
+    'failover_finish': 'failover',
+}
+ANNOTS_EVENT_TAGS = {
+    # 'dataset_created': 'success',
+    # 'dataset_dropped': 'warning',
+
+    'analytics_index_created': 'success',
+    'analytics_index_dropped': 'warning',
+
+    # 'task_finished': 'success',
+    # 'task_started': 'info',
+
+    'backup_removed': 'warning',
+    'backup_paused': 'warning',
+    'backup_resumed': 'info',
+    'backup_plan_created': 'success',
+    'backup_plan_deleted': 'warning',
+    'backup_repo_created': 'success',
+    'backup_repo_deleted': 'warning',
+    'backup_repo_imported': 'success',
+    'backup_repo_archived': 'success',
+
+    'rebalance_start': 'info',
+    'rebalance_finish': 'info',
+
+    'failover_start': 'warning',
+    'failover_end': 'warning',
+
+    'node_joined': 'success',
+    'node_went_down': 'warning',
+
+    # 'eventing_function_deployed': 'success',
+    # 'eventing_function_undeployed': 'warning',
+
+    'fts_index_created': 'success',
+    'fts_index_dropped': 'warning',
+
+    'index_created': 'success',
+    'index_deleted': 'warning',
+    # 'indexer_active': 'info',
+    # 'index_built': 'info',
+
+    'bucket_created': 'success',
+    'bucket_deleted': 'warning',
+    'bucket_updated': 'info',
+    'bucket_flushed': 'warning',
+
+    # 'dropped_ticks': 'info',
+    'data_lost': 'failure',
+    'server_error': 'failure',
+    'sigkill_error': 'failure',
+    'lost_connection_to_server': 'failure',
+
+    # 'LDAP_settings_modified': 'info',
+    # 'password_policy_changed': 'info',
+    # 'group_added': 'success',
+    # 'group_deleted': 'warning',
+    # 'user_added': 'success',
+    # 'user_deleted': 'warning',
+
+    'XDCR_replication_create_started': 'info',
+    'XDCR_replication_remove_started': 'info',
+    'XDCR_replication_create_failed': 'failure',
+    'XDCR_replication_create_successful': 'success',
+    'XDCR_replication_created': 'success',
+    'XDCR_replication_remove_failed': 'failure',
+    'XDCR_replication_remove_successful': 'success',
+}
```

Comment:
It'd probably make sense to put this into a JSON file, but I'm not sure where the best place for this sort of data is in Promtimer.

Reply:
I think it's ok to leave this stuff in annotations.py.